flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -412,7 +412,7 @@ public void registerType(Class<?> type) {
 		if (from > to) {
 			throw new IllegalArgumentException("Start of sequence must not be greater than the end");
 		}
-		return fromCollection(new NumberSequenceIterator(from, to), BasicTypeInfo.LONG_TYPE_INFO, "Sequence Source");
+		return addSource(new StatefulSequenceSource(from, to), "Sequence Source").setParallelism(1);
 	}
```

Comment:
We discussed this quickly and came to the solution that we remove method generateParallelSequence from the StreamExecutionEnvironment since the batch API also only has generateSequence which can be used for "parallelism 1" sources and parallel sources.